bee-common/bee-packable/src/packable/bounded_int.rs

```diff
@@ -0,0 +1,122 @@
+use crate::{
+    coerce::CoerceInfallible,
+    error::{PackError, UnpackError},
+    packer::Packer,
+    unpacker::Unpacker,
+    Packable,
+};
+
+use core::convert::{Infallible, TryFrom, TryInto};
+
+/// Trait that provides an interface for bounded integers.
+pub trait Bounded {
+    /// The type used to define the bounds.
+    type Bounds;
+
+    /// Minimum bounded value.
+    const MIN: Self::Bounds;
+
+    /// Maximum bounded value.
+    const MAX: Self::Bounds;
+
+    /// Actual value.
+    fn value(&self) -> Self::Bounds;
```

Comment:
I wonder if we should just add `Bounded: Into<Self::Bounds>`.

Reply:
Yes, that's a nicer way of doing it.